metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -817,78 +812,78 @@ def _find_dataflow_recipe(
         time_range_constraint: Optional[TimeRangeConstraint] = None,
     ) -> Optional[DataflowRecipe]:
         linkable_specs = linkable_spec_set.as_tuple
-        potential_source_nodes: Sequence[BaseOutput]
-        input_time_spine_source_node: Optional[MetricTimeDimensionTransformNode] = None
+        candidate_nodes_for_left_side_of_join: Sequence[BaseOutput]
+        candidate_nodes_for_right_side_of_join: Sequence[BaseOutput]
         if measure_spec_properties:
-            source_nodes = self._source_nodes
-            potential_source_nodes = self._select_source_nodes_with_measures(
-                measure_specs=set(measure_spec_properties.measure_specs), source_nodes=source_nodes
+            candidate_nodes_for_right_side_of_join = self._source_node_set.source_nodes_for_metric_queries
+            candidate_nodes_for_left_side_of_join = self._select_source_nodes_with_measures(
+                measure_specs=set(measure_spec_properties.measure_specs),
+                source_nodes=self._source_node_set.source_nodes_for_metric_queries,
             )
             default_join_type = SqlJoinType.LEFT_OUTER
         else:
-            # Only read nodes can be source nodes for queries without measures
-            source_nodes = list(self._read_nodes)
-            potential_source_nodes = self._select_read_nodes_with_linkable_specs(
-                linkable_specs=linkable_spec_set, read_nodes=self._read_nodes
-            )
-            # `metric_time` does not exist if there is no metric in the query.
-            # In that case, we'll use the time spine table to represent `metric_time` values.
-            requested_metric_time_specs = [
-                time_dimension_spec
-                for time_dimension_spec in linkable_spec_set.time_dimension_specs
-                if time_dimension_spec.element_name == self._metric_time_dimension_reference.element_name
-            ]
-            if requested_metric_time_specs:
-                input_time_spine_source_node = self._time_spine_source_node
-                # Add time_spine source node to potential source nodes
-                potential_source_nodes = list(potential_source_nodes) + [self._time_spine_source_node]
+            # For the case where we want to find a recipe for a group-by-item and metric_time, we add the time spine
+            # node (which has metric_time for all date values), and set the default join type between nodes to
+            # FULL_OUTER. This will generate a recipe where you join the node containing the group-by-item and the
```

Comment:
This comment is a bit confusing and I think not quite accurate. This `else` clause applies to `all` queries without metrics, but the comment makes it sound like it only applies to queries without metrics that also include `metric_time`.
When querying only group by items, the default join type will _always_ be `FULL OUTER`. This makes it sound like that is only the case when `metric_time` is included in the group by items.

This comment is extra confusing because when `metric_time` is included, the join type is likely going to end up being `CROSS JOIN` (but that probably shouldn't be mentioned here since it is not the _default_ join type, and that logic comes later).

Reply:
Good call, this was an incorrect comment that I added while I was trying to understand the logic. I've removed it.